Documentation/api-guidelines/nullability.md

```diff
@@ -0,0 +1,77 @@
+
+As such, for now, we will start with `Object.ToString` returning `string?`.  We can re-evaluate this decision as we get more experience with consumers of the feature.
+
+In contrast, for `Exception.Message` which is also virtual, we plan to have it be non-nullable, even though technically a derived class could override it to return null, because doing so is so rare that we couldn't find any meaningful examples of doing so.
+
+## Code Review Guidance
+
+Code reviews for enabling the nullability warnings are particularly interesting in that often differ significantly from general code reviews.  Typically, a code reviewer focused only on the code actually being modified, and thus that shows up in a code review diffing tool.  However, enabling the nullability feature has a much broader impact, in that it effectively inverts the meaning of every reference type use in the codebase (or, more specifically, in the scope at which the nullability warning context was applied).  So, for example, if you turn on nullability for a whole file (`#enable nullable` at the top of the file) and then touch no other lines in the file, every method that accepts a `string` is now accepting a non-nullable `string`; whereas previously passing in `null` to that argument would be fine, now the compiler will warn about it, and to allow nulls, the argument must be changed to `string?`.  This means that enabling nullability checking requires reviewing all exposed APIs in that context, regardless of whether they were modified or not, as the contract exposed by the API may have been implicitly modified.
+
+A code review for enabling nullability generally involves two or three passes:
+
+1.	Review all implementation changes made in the code.  Except when explicitly fixing a bug (which should be rare), the annotations employed for nullability should have zero impact on the generated IL (other than potentially some added attributes in the metadata).  The most common changes are:
+a.	Adding `?` to reference type parameters and local symbols.  These inform the compiler that nulls are allowed.  For locals, they evaporate entirely at compile time.  For parameters, they impact the [Nullable(�)] attributes emitted into the metadata by the compiler, but have no effect on the implementation IL.
```

Comment:
Markdown doesn't do bullets with alpha numerics. I suggest to indent and using numbers so that it renders in a readable way.